File: Frontend/src/app/data/dto/receive/profile.dto.ts
import {PictureDto} from "../send/picture.dto";
import {GenderEnum} from "../../../shared/enums/gender.enum";
import {SexualPreferenceEnum} from "../../../shared/enums/SexualPreference.enum";
import {InterestDto} from "./interest.dto";

/**
 * The profile data transfer object received by the client.
 */
export class ProfileDto {

	/**
	 * The user id.
	 */
	userId!: number;

	/**
	 * The gender
	 */
	gender!: GenderEnum;

	/**
	 * The sexual orientation of the user.
	 */
	sexualPreference!: SexualPreferenceEnum;

	/**
	 * The user interests.
	 */
	interests!: InterestDto[];

	/**
	 * The description of the user
	 */
	description!: string;

	/**
	 * The latitude
	 */
	latitude!: number;

	/**
	 * The longitude
	 */
	longitude!: number;

	/**
	 * The city
	 */
	city!: string;

	/**
	 * The fame rating
	 */
	fameRating!: number;

	/**
	 * The date of birth of the user.
	 */
	birthdate!: Date;

	/**
	 * The pictures of the user.
	 */
	pictures: PictureDto[] = [];
}

File: Backend/src/main/java/com/web/matcha/web/dto/ProfileDto.java
package com.web.matcha.web.dto;

import com.web.matcha.domain.enums.GenderEnum;
import com.web.matcha.domain.enums.SexualPreferenceEnum;
import lombok.Getter;
import lombok.Setter;

import java.sql.Timestamp;
import java.util.List;

@Getter
@Setter
public class ProfileDto {

	private Integer id;

	private GenderEnum gender;

	private SexualPreferenceEnum sexualPreference;

	private String description;

	private Timestamp birthdate;

	private LocationDto location;

	private List<String> interests;

	private List<PictureDto> pictures;
}
